subprojects/core/src/main/java/org/gradle/initialization/DefaultGradleLauncher.java

```diff
@@ -201,6 +147,75 @@ public void stop() {
         }
     }
 
+    private class RunBuildAction implements Transformer<BuildResult, BuildOperationContext> {
+        private final Stage upTo;
+
+        private RunBuildAction(Stage upTo) {
+            this.upTo = upTo;
+        }
+
+        @Override
+        public BuildResult transform(BuildOperationContext buildOperationContext) {
+            BuildOperationWorkerRegistry.Completion workerLease = buildServices.get(BuildOperationWorkerRegistry.class).operationStart();
```

Comment:
The JavaDoc of `BuildOperationWorkerRegistry` says "Only use this if you can't use an `Action`. If you can use an `Action`, use `BuildOperationProcessor` instead". So I guess that's what we should do here.